src/api/component.js

```diff
@@ -283,37 +283,70 @@ export default class extends HTMLElement {
       }
     }
 
-    if (attributeChanged) {
+    // DEPRECATED
+    //
+    // static attributeChanged()
+    const { attributeChanged } = this.constructor;
+    if (isFunction(attributeChanged)) {
       attributeChanged(this, { name, newValue, oldValue });
     }
   }
 
   // Skate
   //
-  // Maps to the static updated() callback. That logic should be moved here
-  // when that is finally removed.
-  updatedCallback (prev) {
-    return this.constructor.updated(this, prev);
+  updatedCallback (prevProps) {
+    // DEPRECATED
+    //
+    // static updated()
+    const { updated } = this.constructor;
+    if (isFunction(updated)) {
+      return updated(this, prevProps);
+    }
+
+    // updatedCallback() default implementation
+    //
+    if (!prevProps) {
+      return true;
+    }
+
+    // Use getAllKeys to include all props names and Symbols
+    const allKeys = getAllKeys(prevProps);
+    // Use classic loop because 'for ... of' skips symbols
+    for (let i = 0; i < allKeys.length; i++) {
+      const nameOrSymbol = allKeys[i];
+      if (prevProps[nameOrSymbol] !== this[nameOrSymbol]) {
+        return true;
+      }
+    }
+
+    return false;
   }
 
   // Skate
   //
-  // Maps to the static rendered() callback. That logic should be moved here
-  // when that is finally removed.
   renderedCallback () {
-    return this.constructor.rendered(this);
+    // DEPRECATED
+    //
+    // static rendered()
+    const { rendered } = this.constructor;
+    if (isFunction(rendered)) {
+      return rendered(this);
+    }
   }
 
   // Skate
   //
   // Maps to the static renderer() callback. That logic should be moved here
   // when that is finally removed.
+  // todo: rendererCallback is not documented
```

Comment:
This is somewhat intentional at this point. I'd like to finalise its implementation before officially making it API.